python/pyarrow/includes/libarrow_dataset.pxd

```diff
@@ -18,6 +18,7 @@
 # distutils: language = c++
 
 from libcpp.unordered_map cimport unordered_map
+from libcpp cimport bool
```

Comment:
So this works ([it creates an alias for `bint` named `bool`](https://github.com/cython/cython/blob/d2d2ea33a21e1915a52790ef64fdd1f28867854c/Cython/Includes/libcpp/__init__.pxd#L2)) but elsewhere we either use `bint` directly or call it `c_bool` with: `from libcpp cimport bool as c_bool`.  Lets stick with one of the two existing approaches instead of adding a third.

Reply:
Made the change, thanks!